nextjs-13/src/lib/handler/base.handler.ts

```diff
@@ -0,0 +1,37 @@
+import { User } from '@prisma/client';
+import { NextResponse, NextRequest } from 'next/server';
+import { getToken } from 'next-auth/jwt';
+
+import { findUserById } from 'repositories/user.repository';
+
+export default async function baseHandler(
+  req: NextRequest,
+  callback: (currentUser: User, body: unknown) => Promise<NextResponse>
+) {
+  try {
+    const token = await getToken({ req });
+
+    if (token && token.userId) {
+      const currentUser = await findUserById(token.userId as string);
+
+      if (currentUser) {
+        let body = {};
+
+        try {
+          body = await req.json();
+        } catch (err) {}
+
+        console.log('========>body : ', body);
+        return await callback(currentUser, body);
+      }
+    }
+
+    return NextResponse.json({ message: 'Invalid token' }, { status: 401 });
+  } catch (err) {
+    console.log('========>err : ', err);
```

Comment:
For this I can we can use `console.warn` or `console.error` for better view in the Dev console

Reply:
Improved [here](https://github.com/nimblehq/js-framework-benchmark/pull/96/commits/7e83686ee97e415876c573b274caaf60c3984dd6#diff-add2a84e0fcd856a4db7e6914984152b0d780f178b1988d446d1d0d0a9e7eac6R31)